client/securedrop_client/sdk/__init__.py

```diff
@@ -147,12 +164,102 @@ def _send_json_request(
         if timeout:
             data["timeout"] = timeout
 
+        env = {}
+        if self.development_mode:
+            env["SD_PROXY_ORIGIN"] = self.server
+
+        # Streaming
+        if stream:
+            fobj = tempfile.TemporaryFile("w+b")
+
+            content_length = 0
+            filename = ""
+            sha256sum = ""
+
+            retry = 0
+            bytes_written = 0
+            download_finished = False
+
+            while not download_finished and retry < DOWNLOAD_RETRY_LIMIT:
+                try:
+                    # Open the process
+                    proc = subprocess.Popen(
+                        self._rpc_target(),
+                        stdin=subprocess.PIPE,
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.PIPE,
+                        env=env,
+                    )
+
+                    # Send the data, included the range if we are retrying
+                    if retry > 0:
+                        data["headers"]["Range"] = f"bytes={bytes_written}-"
+
+                    data_str = json.dumps(data).encode()
+                    if proc.stdin is not None:
+                        proc.stdin.write(data_str)
+                        proc.stdin.close()
+
+                    # Save the response headers from the first request
+                    if retry == 0:
+                        try:
+                            if proc.stderr is not None:
+                                stderr = json.loads(proc.stderr.read().decode())
+                                sha256sum = stderr["headers"]["etag"]
+                                filename = stderr["headers"]["content-disposition"]
+                                content_length = int(stderr["headers"]["content-length"])
+                        except (json.decoder.JSONDecodeError, KeyError) as err:
+                            raise BaseError(
+                                "Unable to parse header metadata from response"
+                            ) from err
+
+                    # Write the contents to disk
+                    chunk_size = 64
```

Comment:
Oh, right. I changed it from 1024 bytes down to 64 bytes just so I could confirm that it was actually splitting the files into chunks, since at the moment all of the test files are smaller than 1024 bytes. I forgot to bring it back up, but I agree 1024 is a better chunk size.